### PR TITLE
Various fixes to final stream intel to send in all cases

### PR DESCRIPTION
**Pull request by alyssawilk**
coauthered-by: carloseltuerto 
Most of https://github.com/envoyproxy/envoy-mobile/issues/1520

### CODE REVIEW

**Review by carloseltuerto — APPROVED** [review]

File: library/common/http/client.cc [hunk 1]
```diff
@@ -257,9 +213,7 @@ void Client::DirectStreamCallbacks::onComplete() {
     http_client_.stats().stream_failure_.inc();
   }
 
-  envoy_final_stream_intel final_intel;
-  setFinalStreamIntel(final_intel);
-  bridge_callbacks_.on_complete(streamIntel(), final_intel, bridge_callbacks_.context);
+  bridge_callbacks_.on_complete(streamIntel(), finalStreamIntel(), bridge_callbacks_.context);
```
Comment (junr03) [inline library/common/http/client.cc:216]: We don't latch here because closeStream latches, correct?

File: library/common/http/client.h [hunk 2]
```diff
@@ -37,17 +37,6 @@ struct HttpClientStats {
   ALL_HTTP_CLIENT_STATS(GENERATE_COUNTER_STRUCT)
 };
 
-struct LatencyInfo {
```
Comment (junr03) [inline library/common/http/client.h:40]: nice, one more struct consolidated :)

File: library/common/http/client.cc [hunk 3]
```diff
@@ -286,9 +240,8 @@ void Client::DirectStreamCallbacks::onError() {
             direct_stream_.stream_handle_);
   http_client_.stats().stream_failure_.inc();
 
-  envoy_final_stream_intel final_intel;
-  setFinalStreamIntel(final_intel);
-  bridge_callbacks_.on_error(error_.value(), streamIntel(), final_intel, bridge_callbacks_.context);
+  bridge_callbacks_.on_error(error_.value(), streamIntel(), finalStreamIntel(),
```
Comment (junr03) [inline library/common/http/client.cc:243]: same Q as onComplete

**Review by junr03 — COMMENTED** [review]
lgtm, just a q

File: library/common/http/client.cc (hunk 1 above)
Comment (alyssawilk) [reply library/common/http/client.cc:216]: correct - the stream may be gone at the point onComplete is called

File: library/common/http/client.cc (hunk 3 above)
Comment (alyssawilk) [reply library/common/http/client.cc:243]: yeah we latch on reset which is the only thing that calls onerror

**Review by junr03 — APPROVED** [review]

**Pull request merged**